agent/job_runner.go

```diff
@@ -283,30 +285,55 @@ func (r *JobRunner) Run() error {
 	signal := ""
 	signalReason := ""
 
-	// Run the process. This will block until it finishes.
-	if err := r.process.Run(); err != nil {
-		// Send the error as output
-		r.logStreamer.Process(fmt.Sprintf("%s", err))
+	// Before executing the bootstrap process with the received Job env,
+	// execute the pre-bootstrap hook (if present) for it to tell us
+	// whether it is happy to proceed.
+	environmentCommandOkay := true
+
+	// FIXME(KD) is looking on disk suitable for this use case, do we want an
```

Comment:
Discussing this with @pda I think this is fine without a flag and up to the environment surrounding the agent to provide security.

The directory for the hooks should owned by another unix user that does not provide w or chmod permissions to the user running the buildkite-agent.

Reply:
This is an interesting idea. Is this to perhaps guard against someone removing the pre-bootstrap and thus remove all the security checks?

Reply:
That was my original idea, though as @pda pointed out if they can remove the file they can also replace it with a symlink to `/bin/true` so I don’t think this would actually provide any security benefit :+1:

Reply:
Yeah, I was going to say the same thing :P